src/app/views/query-response/adaptive-cards/AdaptiveCard.tsx

```diff
@@ -0,0 +1,95 @@
+import * as AdaptiveCardsAPI from 'adaptivecards';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
+import { IAdaptiveCardProps } from '../../../../types/adaptivecard';
+import { getAdaptiveCard } from '../../../services/actions/adaptive-cards-action-creator';
+import { Monaco } from '../../common';
+
+class AdaptiveCard extends Component<IAdaptiveCardProps> {
+  private adaptiveCard: AdaptiveCardsAPI.AdaptiveCard;
+  constructor(props: IAdaptiveCardProps) {
+    super(props);
+    this.adaptiveCard = new AdaptiveCardsAPI.AdaptiveCard();
+  }
+
+  public componentDidMount() {
+    const { body , sampleQuery, hostConfig } = this.props;
+    this.props.actions!.getAdaptiveCard(body, sampleQuery);
+    if (hostConfig) {
+      this.adaptiveCard.hostConfig = new AdaptiveCardsAPI.HostConfig(hostConfig);
+    }
+  }
+
+  public componentDidUpdate(nextProps: IAdaptiveCardProps) {
+    const { body , sampleQuery } = this.props;
+    if (JSON.stringify(nextProps.body) !== JSON.stringify(body)) {
+      // we need to update the card as our body has changed
+      this.props.actions!.getAdaptiveCard(body, sampleQuery);
+    }
+  }
+
+  public componentWillUnmount() {
+    // Remove all references
+    delete this.adaptiveCard;
+  }
+
+  public shouldComponentUpdate(nextProps: IAdaptiveCardProps) {
+    if (JSON.stringify(this.props.body) !== JSON.stringify(nextProps.body)) {
+      return true; // body has changed so card will too
+    }
+    if (JSON.stringify(nextProps.card.data) === JSON.stringify(this.props.card.data)) {
+      return false; // card still the same no need to re-render
+    }
+    return true;
+  }
+
+  public render() {
+    const { data , pending } = this.props.card;
+
+    if (pending) {
+      return (<Monaco
+        body={'Fetching Adaptive Card ...'}
+      />);
+    }
+
+    if (data) {
+      try {
+        this.adaptiveCard.parse(data);
```

Comment:
This parsing will be done everytime the component re-renders which might result in a perfomance hit. Perhaps, move this to `componentDidMount`?

Reply:
This is not really an issue as the `componentShouldUpdate` prevents unnecessary renders as discussed.